In [1]:
import gc
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import feather
import datetime
from matplotlib_venn import venn2
from sklearn import preprocessing
from sklearn.model_selection import KFold
from sklearn.model_selection import StratifiedKFold
import lightgbm as lgbm
from sklearn.model_selection import train_test_split
from sklearn.metrics import roc_auc_score
from sklearn.decomposition import LatentDirichletAllocation as LDA
pd.set_option('display.max_Columns', 100)

In [2]:
cartlog = pd.read_feather('../inputs/cartlog.f')
product_master = pd.read_feather('../inputs/product_master.f')
meta = pd.read_feather('../inputs/meta.f')
user_master = pd.read_feather('../inputs/user_master.f')
test = pd.read_csv('../inputs/test.csv')
display_action_id = pd.read_csv('../inputs/display_action_id.csv')

product_master['JAN'] = product_master['JAN'].astype(str)

In [3]:
test_sessions = test["session_id"].unique()
print(len(test_sessions))
test_input_log = cartlog[cartlog["session_id"].isin(test_sessions)]

56486


In [4]:
target_category = [
    38,  # アイスクリーム__ノベルティー
    110,  # スナック・キャンディー__ガム
    113,  # スナック・キャンディー__シリアル
    114,  # スナック・キャンディー__スナック
    134,  # チョコ・ビスクラ__チョコレート
    171,  # ビール系__RTD
    172,  # ビール系__ノンアルコール
    173,  # ビール系__ビール系
    376,  # 和菓子__米菓
    435,  # 大型PET__無糖茶（大型PET）
    467,  # 小型PET__コーヒー（小型PET）
    537,  # 水・炭酸水__大型PET（炭酸水）
    539,  # 水・炭酸水__小型PET（炭酸水）
    629,  # 缶飲料__コーヒー（缶）
    768,  # 麺類__カップ麺
]

In [5]:
# 2020-08-01以前で10分以上経過し購買が発生したセッションにtrainデータを絞る
tmp_sessions = meta[meta['date'] < '2020-08-01']['session_id'].unique()
tmp_log = cartlog[cartlog["session_id"].isin(tmp_sessions)]
print('2020-08-01以前: ', len(tmp_sessions))

# 購買が発生したセッション
payment_sessions = set(tmp_log[tmp_log['is_payment']==1]['session_id'].unique())
print('購買が発生: ', len(payment_sessions))
# 10分以上のセッション
over10min_sessions = set(tmp_log[tmp_log['spend_time']>=600]['session_id'].unique())
print('10分以上: ', len(over10min_sessions))
# 積集合
all_train_sessions = payment_sessions & over10min_sessions
print('積集合: ', len(all_train_sessions))

# 10分以上の全trainのログデータ
all_train_log = tmp_log[tmp_log["session_id"].isin(all_train_sessions)]
print('全trainのログ: ', len(all_train_log))

2020-08-01以前:  663708
購買が発生:  618462
10分以上:  404825
積集合:  391383
全trainのログ:  10826062


In [6]:
def agg_payment(cartlog) -> pd.DataFrame:
    """セッションごと・商品ごとの購買個数を集計する"""
    # JANコード (vale_1)ごとに商品の購入個数(n_items)を足し算
    agg = cartlog.loc[cartlog["kind_1"] == "商品"].groupby(["session_id", "value_1"])["n_items"].sum().reset_index()
    agg = agg.rename(columns={"value_1": "JAN"})
    return pd.merge(agg, product_master[["JAN", "category_id"]], on="JAN", how="inner").drop(columns=['JAN'])

In [7]:
class RetailDataset:
    def __init__(self, thres_sec, meta):
        self.thres_sec = thres_sec
        self.meta = meta.copy()
        self.meta['time_elapsed_sec'] = self.meta['time_elapsed'] * 60
        self.meta.loc[self.meta['time_elapsed_sec'].isnull(), 'time_elapsed_sec'] = thres_sec
        
        # all_train_logの中で、指定時間より前のログのみを抽出　-> public_train_log, train_sessions
        merge_train = pd.merge(all_train_log, self.meta[["session_id", "time_elapsed_sec"]], on=["session_id"], how="inner")
        self.public_train_log = merge_train[merge_train['spend_time'] <= merge_train['time_elapsed_sec']]
        self.train_sessions = self.public_train_log["session_id"].unique()
        
    def get_train_input_log(self) -> pd.DataFrame:
        return self.public_train_log

    def get_train_target(self) -> pd.DataFrame:
        """学習で使用するセッションの目的変数を取得する"""
        train_target = pd.DataFrame(
            index=self.train_sessions,
        )
        train_target.index.name = "session_id"

        # time_elapsed以降のデータから購買個数を集計する
        merge_train = pd.merge(all_train_log, self.meta[["session_id", "time_elapsed_sec"]], on=["session_id"], how="inner")
        after_elapsed_log = merge_train[merge_train['spend_time'] > merge_train['time_elapsed_sec']]
        
        train_item_num = agg_payment(after_elapsed_log)
        train_item_num = train_item_num[train_item_num['category_id'].isin(target_category)]
        train_target_pos = train_item_num.groupby(["session_id", "category_id"])["n_items"].sum().unstack().fillna(0).astype(int)
        train_target_pos[train_target_pos > 0] = 1
        train_target_pos[train_target_pos <= 0] = 0

        return train_target.join(train_target_pos).fillna(0).reset_index()

In [8]:
def get_train_log(elapsed_min):
    dataset = RetailDataset(elapsed_min*60, meta)
    train_input_log = dataset.get_train_input_log()
    y_train = dataset.get_train_target()
    print('train_session', y_train.shape)
    return train_input_log, y_train

In [9]:
ELAPSED_MIN = [0, 3, 5, 10]
# ELAPSED_MIN = [5]

In [10]:
def save_train_log():
    for elap_min in ELAPSED_MIN:
        train_log, train_y = get_train_log(elap_min)
        train_log = train_log.reset_index(drop=True)
        train_y = train_y.reset_index(drop=True)
        train_y.columns = [str(c) for c in train_y.columns]
        train_log.to_feather('../inputs/train2_log_{}.f'.format(elap_min))
        train_y.to_feather('../inputs/train2_y_{}.f'.format(elap_min))

In [11]:
def load_train_log():
    train_log = {}
    train_y = {}
    for elap_min in ELAPSED_MIN:
        log = pd.read_feather('../inputs/train2_log_{}.f'.format(elap_min))
        y = pd.read_feather('../inputs/train2_y_{}.f'.format(elap_min))
        train_log[elap_min] = log
        train_y[elap_min] = y
    return train_log, train_y

In [12]:
# save_train_log()

- 0, train_session (378594, 16)
- 3, train_session (389649, 16)
- 5, train_session (390621, 16)
- 10, train_session (391074, 16)

### 過去のログデータ

In [13]:
payed_item = agg_payment(all_train_log)

#### クーポン発行とターゲットカテゴリの購入数の関係

In [14]:
def get_coupon_log_data(payed_item):
    # train用のデータから購買した商品
    payed_item_in_cat = payed_item[payed_item['category_id'].isin(target_category)]
    cat_pivot = payed_item_in_cat.pivot_table(
        index='session_id', columns='category_id', values='n_items').fillna(0).add_prefix('coupon_').reset_index()

    # 発券されたクーポン
    coupon = all_train_log[all_train_log['kind_1'] == 'クーポン']
    coupon_count = coupon['value_1'].value_counts()
    print(len(coupon_count))
    session_coupon = coupon[coupon['value_1'].isin(list(coupon_count.index))][['session_id', 'value_1']]
    session_coupon = session_coupon.rename(columns={'value_1':'coupon'})
    session_coupon = session_coupon.groupby(['session_id', 'coupon']).size().reset_index()

    session_cat_coupon = pd.merge(cat_pivot, session_coupon[['session_id', 'coupon']], on='session_id', how='left')
    session_cat_coupon['coupon'] = session_cat_coupon['coupon'].fillna(0)
    session_cat_coupon.drop(columns=['session_id'], inplace=True)
    group_coupon = session_cat_coupon.groupby(['coupon']).mean()
    group_coupon[group_coupon < 0] = 0
    return group_coupon.reset_index()

In [15]:
def LDA_topic(df_input, topic, index, prefix):
    df_cp = df_input.set_index(index)
    lda = LDA(n_components=topic)
    lda_out = pd.DataFrame(lda.fit_transform(df_cp), index=df_cp.index).add_prefix(prefix)
    return lda_out.reset_index()

In [16]:
# group_coupon = get_coupon_log_data(payed_item)
# lda_coupon = LDA_topic(group_coupon, 6, 'coupon', 'LDA_coup_')

### ユーザ情報

In [17]:
user_features = pd.merge(meta[["session_id", "user_id"]], user_master, on="user_id", how="left")
user_features.loc[user_features['age'] >= 80, 'age'] = np.NaN
user_features.loc[user_features['age'] < 10, 'age'] = np.NaN
user_features.loc[user_features['gender'] > 1, 'gender'] = np.NaN

In [18]:
def get_user_item(payed_item):
    # train用のデータから購買した商品
    user_payed_item = pd.merge(payed_item, meta[['session_id', 'user_id']], on='session_id', how='left')
    group_user_item = user_payed_item.groupby(['user_id', 'category_id'])[['n_items']].sum().reset_index()
    pivot_user_item = group_user_item.pivot_table(index='user_id', columns='category_id', values='n_items')
    # 全ユーザーの購入数の合計が5000以上のカテゴリに絞り込み
    sum_user_item = pivot_user_item.sum()
    user_item_index = sum_user_item[sum_user_item > 5000].index
    pivot_user_item = pivot_user_item[user_item_index].fillna(0).reset_index()
    # trainに存在しない人用に平均値で穴埋め
    user_item_mean = pivot_user_item.mean()
    # 全ユーザーとマージ
    all_user_item = pd.merge(user_master[['user_id']], pivot_user_item, on='user_id', how='left')
    # targetのカテゴリは除く
    for col in all_user_item.columns:
        if (col == 'user_id') or (col in target_category):
            continue
        new_col = 'user_pay_{}'.format(col)
        all_user_item[new_col] = all_user_item[col].fillna(user_item_mean[col]).astype('float32')
        all_user_item.loc[all_user_item[new_col]<0, new_col] = 0

    return all_user_item.drop(columns=list(user_item_mean.index))

In [19]:
all_user_item = get_user_item(payed_item)
print(all_user_item.shape)

(40350, 231)


In [20]:
# %%time
# user_lda = LDA_topic(all_user_item, 10, 'user_id', 'LDA_')
# user_lda.to_feather('../inputs/user_lda.f')

In [21]:
user_lda = pd.read_feather('../inputs/user_lda.f')
all_user_item = pd.merge(all_user_item, user_lda, on='user_id', how='left')

In [22]:
print(all_user_item.shape)

(40350, 241)


### メタ情報

In [23]:
def get_meta_features(meta):
    meta_features = meta.copy()
    meta_features['year'] = meta_features['date'].dt.year
    meta_features['month'] = meta_features['date'].dt.month
    meta_features['day'] = meta_features['date'].dt.day
    meta_features['dow'] = meta_features['date'].dt.dayofweek
    meta_features['doy'] = meta_features['date'].dt.dayofyear
    meta_features['week_time'] = meta_features['dow'] * 24 + meta_features['hour']
    le = preprocessing.LabelEncoder()
    meta_features['userid'] = le.fit_transform(meta_features['user_id'])
    
    # 曜日x時間の来店者数
    df_tz = meta_features.groupby(['week_time']).size().rename('timezone_count')
    df_tz =  pd.DataFrame(df_tz).reset_index()
    meta_features = pd.merge(meta_features, df_tz, on='week_time', how='left')
    
    # userごとに前の来店からどれくらいの日数が経ったか
    df = meta_features.groupby(['session_id', 'user_id', 'date']).first().reset_index().sort_values(['user_id', 'session_id'])
    df['date_diff'] = df['date'].diff(1)
    df['user_diff'] = df['user_id'].shift(1)
    df.loc[df['user_diff'] != df['user_id'], 'date_diff'] = pd.NaT
    df['date_diff'] = df['date_diff'].dt.days
    meta_features = meta_features.join(df[['date_diff']])

    # userごとの来店回数
    meta_features['date_rank'] = meta_features.groupby(['user_id'])['date'].rank(ascending=True)
    
    return meta_features.drop(columns=['user_id', 'date', 'time_elapsed', 'date_str'])

In [24]:
meta_features = get_meta_features(meta)

### ディスプレイアクション

In [25]:
disp_name_dic = {}
for i, disp in enumerate(display_action_id['display_name'].unique()):
    disp_name_dic[disp] = 'disp_cnt_{}'.format(i)
    
act_name_dic = {}
for i, action in enumerate(display_action_id['action_name'].unique()):
    act_name_dic[action] = 'act_cnt_{}'.format(i)

In [26]:
def get_display_name_feature(input_log):
    merge = pd.merge(input_log, display_action_id, on='display_action_id', how='left')
    disp_group_count = merge.groupby(['session_id', 'display_name']).size().reset_index().rename(columns={0:'disp_name_count'})
    disp_name_pivot = disp_group_count.pivot_table(index='session_id', columns='display_name', values='disp_name_count', aggfunc='sum')
    disp_name_pivot = disp_name_pivot.reset_index().fillna(0).rename(columns=disp_name_dic)
    
    disp_out = disp_name_pivot[['session_id']].copy()
    for val in disp_name_dic.values():
        disp_out[val] = 0
    
    for col in disp_name_pivot.columns:
        if col == 'session_id':
            continue
        disp_out[col] = disp_name_pivot[col]
    
    return disp_out

In [27]:
def get_action_name_feature(input_log):
    merge = pd.merge(input_log, display_action_id, on='display_action_id', how='left')
    act_group_count = merge.groupby(['session_id', 'action_name']).size().reset_index().rename(columns={0:'act_name_count'})
    act_name_pivot = act_group_count.pivot_table(index='session_id', columns='action_name', values='act_name_count', aggfunc='sum')
    act_name_pivot = act_name_pivot.reset_index().fillna(0).rename(columns=act_name_dic)
    
    act_out = act_name_pivot[['session_id']].copy()
    for val in act_name_dic.values():
        act_out[val] = 0
    
    for col in act_name_pivot.columns:
        if col == 'session_id':
            continue
        act_out[col] = act_name_pivot[col]
    
    return act_out

### セッション単位の特徴量

In [28]:
# def get_coupon_info(input_log):
#     session_coupon = input_log[input_log["kind_1"] == "クーポン"][['session_id', 'value_1']].rename(columns={'value_1':'coupon'})
#     session_coupon = pd.merge(session_coupon, lda_coupon, on='coupon', how='left').drop(columns=['coupon'])
#     session_coupon = session_coupon.groupby(['session_id']).max().reset_index()
#     return session_coupon

In [29]:
def get_pre_payment_item(input_log):
    session_unique = input_log['session_id'].unique()
    agg = input_log.loc[input_log["kind_1"] == "商品"].groupby(["session_id", "value_1"])["n_items"].sum().reset_index()
    agg = agg.rename(columns={"value_1": "JAN"})
    agg = pd.merge(agg, product_master[["JAN", "category_id"]], on="JAN", how="inner")
    agg = agg[agg['category_id'].isin(target_category)]
    agg = agg.groupby(["session_id", "category_id"])["n_items"].sum().reset_index()
    
    sesi = np.zeros(len(target_category))
    cate = [ct for ct in target_category]
    
    dummy = pd.DataFrame({'session_id':sesi, 'category_id':cate, 'n_items':sesi})
    agg = pd.concat([agg, dummy])
    
    agg = agg.pivot_table(index='session_id', columns='category_id', values='n_items').fillna(0)
    src_columns = ['x_{}'.format(c) for c in agg.columns]
    agg.columns = src_columns
    
    col = ['pre_target_{}'.format(c) for c in target_category]
    df_out = pd.DataFrame(index=session_unique, columns=col)
    df_out.index.name = "session_id"
    df_out = df_out.join(agg)
    for ct in target_category:
        src = 'x_{}'.format(ct)
        dst = 'pre_target_{}'.format(ct)
        df_out[dst] = df_out[src]
    
    return df_out.drop(columns=src_columns).fillna(0).reset_index()

In [30]:
def get_session_kind_group(input_log):
    kind_name ={
        'クーポン': 'coupon',
        '会計': 'kaikei',
        'キー': 'key',
        'カテゴリ': 'categry',
        'バーコードスキャン': 'barcode',
        'UUID': 'uuid',
        '使用ポイント': 'usedpoint',
        '確認': 'confirm',
        'ブランドスイッチ': 'bland',
        'レシピ': 'recipe',
        'スマホスキャン': 'smartphone',
        '磁気スキャン': 'magnetic',
        'レコメンド': 'recommend',
        '倍率ポイント': 'point',
    }
    group_count = input_log[input_log['kind_1'] == '商品'].groupby(["session_id"]).size().rename('group_count_'+'item')
    for kind, name in kind_name.items():
        tmp = input_log[input_log['kind_1'] == kind].groupby(["session_id"]).size().rename('group_count_'+name)
        group_count = pd.concat([group_count, tmp], axis=1)
        
    return group_count.reset_index()

In [31]:
def get_session_info(input_log):
    # アクション数
    n_actions = input_log.groupby(["session_id"]).size().rename("n_actions")
    # 経過時間の平均
    mean_spend_time = input_log.groupby(["session_id"])["spend_time"].mean()
    
    session_features = pd.concat([
        n_actions,
        mean_spend_time,
    ], axis=1)
    return session_features.reset_index()

### セッション単位で集計

In [32]:
def get_session_features(input_log):
    df_ses = pd.DataFrame(columns=['session_id'])
    
    session_feat = [
        get_session_info(input_log),
        get_session_kind_group(input_log),
        get_display_name_feature(input_log),
        get_action_name_feature(input_log),
        get_pre_payment_item(input_log),
#         get_coupon_info(input_log),
    ]
    
    for feat in session_feat:
        df_ses = pd.merge(df_ses, feat, on='session_id', how='outer')
        
    return df_ses

### 特徴量を集約する

In [33]:
def merge_features(input_log, session):
    feat_list = [
        get_session_features(input_log),
        user_features,
        meta_features,
    ]
    out = pd.DataFrame({"session_id": session})
    for feat in feat_list:
        out = pd.merge(out, feat, on="session_id", how="left")
        
    # userの情報
    out = pd.merge(out, all_user_item, on='user_id', how='left').drop(columns='user_id')

    assert len(session) == len(out)
    return out

In [34]:
def get_train_all_features(elapsed_min, train_log_list, train_y_list):
    train_input_log = train_log_list[elapsed_min]
    y_train = train_y_list[elapsed_min]
    
    train_features = merge_features(train_input_log, y_train['session_id'])
    print('train_features', train_features.shape)
    return train_features, y_train

In [35]:
def get_test_all_feature(elapsed_min):
    test_meta = meta[meta['session_id'].isin(test_sessions)]
    test_meta = test_meta[test_meta['time_elapsed'] == elapsed_min]
    test_input_elapsed = pd.merge(test_input_log, test_meta[['session_id']], on='session_id', how='left')
    
    test_features = merge_features(test_input_elapsed, test_meta['session_id'])
    print('test_features', test_features.shape)
    return test_features

In [36]:
n_fold = 4

In [37]:
lgbm_param = {
    'objective' : 'binary',
    'boosting_type': 'gbdt',
    'metric': 'auc',
    'seed' : 0,
    'learning_rate':  0.1,
#   'max_depth': 6,
    'feature_fraction': 0.6,
    'bagging_fraction': 0.6,
    'bagging_freq': 2,
    'verbose': -1
}

In [38]:
def train_lgbm(X, y, params=lgbm_param):

    fold = StratifiedKFold(n_splits=n_fold, shuffle=True, random_state=0)
    cv = fold.split(X, y)
    
    models = []
    oof_pred = np.zeros_like(y, dtype=np.float)
    
    cat_feat = ['age', 'gender', 'dow']

    for i, (idx_train, idx_valid) in enumerate(cv): 
        x_train, y_train = X.iloc[idx_train], y.iloc[idx_train]
        x_valid, y_valid = X.iloc[idx_valid], y.iloc[idx_valid]
        
        lgbm_train = lgbm.Dataset(x_train, y_train, categorical_feature = cat_feat)
        lgbm_eval = lgbm.Dataset(x_valid, y_valid, reference=lgbm_train, categorical_feature = cat_feat)
        
        lgbm_model = lgbm.train(params, 
                                                    lgbm_train, 
                                                    valid_sets=lgbm_eval,
                                                    categorical_feature = cat_feat,
                                                    num_boost_round=10000,
                                                    early_stopping_rounds=100,
                                                    verbose_eval=-1)
        y_pred = lgbm_model.predict(x_valid, num_iteration=lgbm_model.best_iteration)
        
        oof_pred[idx_valid] = y_pred
        models.append(lgbm_model)

    score = roc_auc_score(y, oof_pred)
    print('--- FINISHED \ whole score: {:.4f} ---'.format(score))
    return oof_pred, models, score

In [39]:
def predict(models, feature):
    pred_list = []
    for i, model in enumerate(models):
        pred = model.predict(feature, num_iteration = model.best_iteration)
        pred_list.append(pred)
    
    score = np.mean(pred_list, axis=0)
    return score

In [40]:
train_log_list, train_y_list = load_train_log()

In [41]:
gc.collect()

0

In [42]:
%%time
df_pred_all = pd.DataFrame()
df_score_all = pd.DataFrame(index=ELAPSED_MIN)
models_list_list = []

for elapsed_min in ELAPSED_MIN:
    print(f'===== {elapsed_min} =====')
    train_features, y_train = get_train_all_features(elapsed_min, train_log_list, train_y_list)
    test_features = get_test_all_feature(elapsed_min)

    df_pred = pd.DataFrame(index=test_features['session_id'])
    train_features.drop(columns=['session_id'], inplace=True)
    test_features.drop(columns=['session_id'], inplace=True)
    
    models_list = []
    for target in y_train.columns:
        if target == 'session_id':
            continue
        
        print(f"---- id = {target} -----")
        oof, models, score = train_lgbm(train_features, y_train[target])
        models_list.append(models)

        pred = predict(models, test_features)
        df_pred[target] = pred
        df_score_all.loc[elapsed_min, target] = score
        
    models_list_list.append(models_list)
    df_pred_all = pd.concat([df_pred_all, df_pred])
    print(len(df_pred_all))

===== 0 =====
train_features (378594, 333)
test_features (14277, 333)
---- id = 38 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[654]	valid_0's auc: 0.822819
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[764]	valid_0's auc: 0.826812
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[671]	valid_0's auc: 0.827768
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[751]	valid_0's auc: 0.827548
--- FINISHED \ whole score: 0.8262 ---
---- id = 110 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[295]	valid_0's auc: 0.791328
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[197]	valid_0's auc: 0.788631
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[223]	valid_0's auc: 0.796906
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[261]	valid_0's auc: 0.800658
--- FINISHED \ whole score: 0.7940 ---
---- id = 113 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[628]	valid_0's auc: 0.804497
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[437]	valid_0's auc: 0.798895
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[557]	valid_0's auc: 0.803115
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[706]	valid_0's auc: 0.802842
--- FINISHED \ whole score: 0.8022 ---
---- id = 114 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[942]	valid_0's auc: 0.757075
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1003]	valid_0's auc: 0.761061
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1144]	valid_0's auc: 0.757511
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[788]	valid_0's auc: 0.759546
--- FINISHED \ whole score: 0.7588 ---
---- id = 134 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[581]	valid_0's auc: 0.744891
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[555]	valid_0's auc: 0.745481
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[668]	valid_0's auc: 0.747706
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[565]	valid_0's auc: 0.743478
--- FINISHED \ whole score: 0.7454 ---
---- id = 171 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[993]	valid_0's auc: 0.867673
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1076]	valid_0's auc: 0.86746
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1321]	valid_0's auc: 0.869322
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1136]	valid_0's auc: 0.87083
--- FINISHED \ whole score: 0.8688 ---
---- id = 172 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[682]	valid_0's auc: 0.869289
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[821]	valid_0's auc: 0.871048
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[635]	valid_0's auc: 0.87223
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[722]	valid_0's auc: 0.869625
--- FINISHED \ whole score: 0.8704 ---
---- id = 173 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1152]	valid_0's auc: 0.865541
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1163]	valid_0's auc: 0.863755
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1040]	valid_0's auc: 0.863412
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1124]	valid_0's auc: 0.864513
--- FINISHED \ whole score: 0.8643 ---
---- id = 376 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[381]	valid_0's auc: 0.752861
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[568]	valid_0's auc: 0.759798
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[679]	valid_0's auc: 0.754312
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[391]	valid_0's auc: 0.753208
--- FINISHED \ whole score: 0.7550 ---
---- id = 435 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[804]	valid_0's auc: 0.864502
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1372]	valid_0's auc: 0.869028
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1010]	valid_0's auc: 0.869568
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1091]	valid_0's auc: 0.871613
--- FINISHED \ whole score: 0.8685 ---
---- id = 467 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[715]	valid_0's auc: 0.872395
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[515]	valid_0's auc: 0.869884
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[294]	valid_0's auc: 0.867
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[460]	valid_0's auc: 0.870935
--- FINISHED \ whole score: 0.8694 ---
---- id = 537 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[620]	valid_0's auc: 0.909763
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[871]	valid_0's auc: 0.901912
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[992]	valid_0's auc: 0.90265
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[448]	valid_0's auc: 0.904608
--- FINISHED \ whole score: 0.9034 ---
---- id = 539 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[892]	valid_0's auc: 0.857118
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[741]	valid_0's auc: 0.858121
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[868]	valid_0's auc: 0.859109
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[715]	valid_0's auc: 0.858025
--- FINISHED \ whole score: 0.8580 ---
---- id = 629 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[632]	valid_0's auc: 0.910037
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[294]	valid_0's auc: 0.918026
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[337]	valid_0's auc: 0.925989
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[489]	valid_0's auc: 0.914256
--- FINISHED \ whole score: 0.9147 ---
---- id = 768 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[798]	valid_0's auc: 0.748477
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[808]	valid_0's auc: 0.746482
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[719]	valid_0's auc: 0.745598
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[789]	valid_0's auc: 0.748294
--- FINISHED \ whole score: 0.7472 ---
14277
===== 3 =====
train_features (389649, 333)
test_features (11304, 333)
---- id = 38 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[643]	valid_0's auc: 0.824434
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[798]	valid_0's auc: 0.821625
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1156]	valid_0's auc: 0.822536
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[610]	valid_0's auc: 0.82713
--- FINISHED \ whole score: 0.8235 ---
---- id = 110 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[161]	valid_0's auc: 0.786698
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[292]	valid_0's auc: 0.784544
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[212]	valid_0's auc: 0.784127
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[148]	valid_0's auc: 0.785067
--- FINISHED \ whole score: 0.7839 ---
---- id = 113 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[401]	valid_0's auc: 0.801775
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[331]	valid_0's auc: 0.787743
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[659]	valid_0's auc: 0.799467
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[692]	valid_0's auc: 0.797812
--- FINISHED \ whole score: 0.7963 ---
---- id = 114 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1104]	valid_0's auc: 0.754969
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[924]	valid_0's auc: 0.75649
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[655]	valid_0's auc: 0.754362
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1008]	valid_0's auc: 0.753339
--- FINISHED \ whole score: 0.7547 ---
---- id = 134 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[721]	valid_0's auc: 0.739572
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[419]	valid_0's auc: 0.740796
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[437]	valid_0's auc: 0.738386
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[494]	valid_0's auc: 0.738967
--- FINISHED \ whole score: 0.7394 ---
---- id = 171 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1144]	valid_0's auc: 0.861397
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1469]	valid_0's auc: 0.860168
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1303]	valid_0's auc: 0.864054
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1371]	valid_0's auc: 0.863223
--- FINISHED \ whole score: 0.8622 ---
---- id = 172 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[741]	valid_0's auc: 0.867629
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[951]	valid_0's auc: 0.85887
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[853]	valid_0's auc: 0.864828
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[708]	valid_0's auc: 0.8621
--- FINISHED \ whole score: 0.8629 ---
---- id = 173 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[978]	valid_0's auc: 0.859035
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[805]	valid_0's auc: 0.854888
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1028]	valid_0's auc: 0.859444
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1096]	valid_0's auc: 0.858677
--- FINISHED \ whole score: 0.8580 ---
---- id = 376 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[550]	valid_0's auc: 0.747302
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[691]	valid_0's auc: 0.751895
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[539]	valid_0's auc: 0.752929
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[430]	valid_0's auc: 0.753263
--- FINISHED \ whole score: 0.7513 ---
---- id = 435 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[637]	valid_0's auc: 0.855545
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[875]	valid_0's auc: 0.852479
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1000]	valid_0's auc: 0.855597
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[872]	valid_0's auc: 0.853779
--- FINISHED \ whole score: 0.8542 ---
---- id = 467 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[639]	valid_0's auc: 0.85762
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[730]	valid_0's auc: 0.86633
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[530]	valid_0's auc: 0.860303
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[593]	valid_0's auc: 0.8584
--- FINISHED \ whole score: 0.8606 ---
---- id = 537 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[926]	valid_0's auc: 0.899048
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[819]	valid_0's auc: 0.897332
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[894]	valid_0's auc: 0.903158
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[706]	valid_0's auc: 0.8971
--- FINISHED \ whole score: 0.8990 ---
---- id = 539 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[971]	valid_0's auc: 0.851168
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[762]	valid_0's auc: 0.850082
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[914]	valid_0's auc: 0.850771
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[929]	valid_0's auc: 0.84788
--- FINISHED \ whole score: 0.8499 ---
---- id = 629 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[209]	valid_0's auc: 0.904101
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[411]	valid_0's auc: 0.912909
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[472]	valid_0's auc: 0.908998
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[452]	valid_0's auc: 0.915563
--- FINISHED \ whole score: 0.9094 ---
---- id = 768 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[628]	valid_0's auc: 0.736392
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[607]	valid_0's auc: 0.736292
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[382]	valid_0's auc: 0.73362
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[491]	valid_0's auc: 0.735278
--- FINISHED \ whole score: 0.7354 ---
25581
===== 5 =====
train_features (390621, 333)
test_features (14072, 333)
---- id = 38 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[501]	valid_0's auc: 0.825451
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[835]	valid_0's auc: 0.820134
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[869]	valid_0's auc: 0.821604
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[714]	valid_0's auc: 0.824065
--- FINISHED \ whole score: 0.8226 ---
---- id = 110 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[190]	valid_0's auc: 0.785008
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[151]	valid_0's auc: 0.77603
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[164]	valid_0's auc: 0.77227
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[176]	valid_0's auc: 0.766898
--- FINISHED \ whole score: 0.7750 ---
---- id = 113 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[613]	valid_0's auc: 0.791509
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[364]	valid_0's auc: 0.793304
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[310]	valid_0's auc: 0.785514
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[447]	valid_0's auc: 0.788989
--- FINISHED \ whole score: 0.7895 ---
---- id = 114 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[704]	valid_0's auc: 0.749244
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[812]	valid_0's auc: 0.751
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1010]	valid_0's auc: 0.748389
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[924]	valid_0's auc: 0.749794
--- FINISHED \ whole score: 0.7496 ---
---- id = 134 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[444]	valid_0's auc: 0.734944
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[649]	valid_0's auc: 0.736919
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[562]	valid_0's auc: 0.736769
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[607]	valid_0's auc: 0.740179
--- FINISHED \ whole score: 0.7372 ---
---- id = 171 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1096]	valid_0's auc: 0.854996
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1381]	valid_0's auc: 0.857074
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1461]	valid_0's auc: 0.853426
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[912]	valid_0's auc: 0.856382
--- FINISHED \ whole score: 0.8552 ---
---- id = 172 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[646]	valid_0's auc: 0.857011
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[747]	valid_0's auc: 0.861599
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[658]	valid_0's auc: 0.85046
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[374]	valid_0's auc: 0.85948
--- FINISHED \ whole score: 0.8556 ---
---- id = 173 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1012]	valid_0's auc: 0.853701
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1378]	valid_0's auc: 0.85271
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1098]	valid_0's auc: 0.854443
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[927]	valid_0's auc: 0.85118
--- FINISHED \ whole score: 0.8529 ---
---- id = 376 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[380]	valid_0's auc: 0.746169
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[660]	valid_0's auc: 0.751517
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[599]	valid_0's auc: 0.750432
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[493]	valid_0's auc: 0.747332
--- FINISHED \ whole score: 0.7489 ---
---- id = 435 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[631]	valid_0's auc: 0.852438
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[879]	valid_0's auc: 0.844102
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[996]	valid_0's auc: 0.84927
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[484]	valid_0's auc: 0.85098
--- FINISHED \ whole score: 0.8484 ---
---- id = 467 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[379]	valid_0's auc: 0.856398
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[484]	valid_0's auc: 0.859947
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[362]	valid_0's auc: 0.854281
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[376]	valid_0's auc: 0.853531
--- FINISHED \ whole score: 0.8560 ---
---- id = 537 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[812]	valid_0's auc: 0.890025
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[905]	valid_0's auc: 0.894522
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[589]	valid_0's auc: 0.893481
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[640]	valid_0's auc: 0.88975
--- FINISHED \ whole score: 0.8914 ---
---- id = 539 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1067]	valid_0's auc: 0.84822
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1044]	valid_0's auc: 0.835019
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[778]	valid_0's auc: 0.843691
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[716]	valid_0's auc: 0.841577
--- FINISHED \ whole score: 0.8417 ---
---- id = 629 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[337]	valid_0's auc: 0.897233
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[285]	valid_0's auc: 0.897793
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[166]	valid_0's auc: 0.90148
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[91]	valid_0's auc: 0.893875
--- FINISHED \ whole score: 0.8942 ---
---- id = 768 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[540]	valid_0's auc: 0.734628
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[338]	valid_0's auc: 0.727901
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[615]	valid_0's auc: 0.735117
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[512]	valid_0's auc: 0.726898
--- FINISHED \ whole score: 0.7311 ---
39653
===== 10 =====
train_features (391074, 333)
test_features (16833, 333)
---- id = 38 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[674]	valid_0's auc: 0.822015
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[899]	valid_0's auc: 0.819584
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[632]	valid_0's auc: 0.822719
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[702]	valid_0's auc: 0.81895
--- FINISHED \ whole score: 0.8207 ---
---- id = 110 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[102]	valid_0's auc: 0.770737
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[245]	valid_0's auc: 0.774802
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[353]	valid_0's auc: 0.77576
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[135]	valid_0's auc: 0.763919
--- FINISHED \ whole score: 0.7673 ---
---- id = 113 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[177]	valid_0's auc: 0.778643
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[460]	valid_0's auc: 0.787091
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[700]	valid_0's auc: 0.796604
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[319]	valid_0's auc: 0.789299
--- FINISHED \ whole score: 0.7866 ---
---- id = 114 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[705]	valid_0's auc: 0.754742
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[947]	valid_0's auc: 0.756863
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[710]	valid_0's auc: 0.756446
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[876]	valid_0's auc: 0.756626
--- FINISHED \ whole score: 0.7562 ---
---- id = 134 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[302]	valid_0's auc: 0.741454
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[375]	valid_0's auc: 0.745103
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[430]	valid_0's auc: 0.746277
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[419]	valid_0's auc: 0.745926
--- FINISHED \ whole score: 0.7447 ---
---- id = 171 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[961]	valid_0's auc: 0.845852
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[756]	valid_0's auc: 0.8393
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[962]	valid_0's auc: 0.841063
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[990]	valid_0's auc: 0.843218
--- FINISHED \ whole score: 0.8423 ---
---- id = 172 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[487]	valid_0's auc: 0.839683
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[218]	valid_0's auc: 0.812979
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[685]	valid_0's auc: 0.854257
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[263]	valid_0's auc: 0.833907
--- FINISHED \ whole score: 0.8333 ---
---- id = 173 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1016]	valid_0's auc: 0.845477
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[907]	valid_0's auc: 0.848587
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[1011]	valid_0's auc: 0.84904
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[619]	valid_0's auc: 0.846719
--- FINISHED \ whole score: 0.8472 ---
---- id = 376 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[582]	valid_0's auc: 0.759792
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[483]	valid_0's auc: 0.754736
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[377]	valid_0's auc: 0.764001
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[512]	valid_0's auc: 0.757879
--- FINISHED \ whole score: 0.7590 ---
---- id = 435 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[760]	valid_0's auc: 0.840854
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[288]	valid_0's auc: 0.835615
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[675]	valid_0's auc: 0.840876
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[368]	valid_0's auc: 0.83527
--- FINISHED \ whole score: 0.8371 ---
---- id = 467 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[333]	valid_0's auc: 0.836442
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[446]	valid_0's auc: 0.831585
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[439]	valid_0's auc: 0.852262
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[127]	valid_0's auc: 0.835007
--- FINISHED \ whole score: 0.8365 ---
---- id = 537 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[427]	valid_0's auc: 0.87587
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[717]	valid_0's auc: 0.875523
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[632]	valid_0's auc: 0.882941
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[612]	valid_0's auc: 0.877944
--- FINISHED \ whole score: 0.8773 ---
---- id = 539 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[591]	valid_0's auc: 0.826668
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[550]	valid_0's auc: 0.826236
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[791]	valid_0's auc: 0.833401
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[904]	valid_0's auc: 0.838722
--- FINISHED \ whole score: 0.8309 ---
---- id = 629 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[632]	valid_0's auc: 0.873558
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[352]	valid_0's auc: 0.880958
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[417]	valid_0's auc: 0.86051
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[295]	valid_0's auc: 0.885268
--- FINISHED \ whole score: 0.8710 ---
---- id = 768 -----


/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
/home/umhrysnb/anaconda3/lib/python3.8/site-packages/lightgbm/basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))


Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[528]	valid_0's auc: 0.730388
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[495]	valid_0's auc: 0.733396
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[387]	valid_0's auc: 0.725714
Training until validation scores don't improve for 100 rounds
Early stopping, best iteration is:
[521]	valid_0's auc: 0.734113
--- FINISHED \ whole score: 0.7309 ---
56486
CPU times: user 1d 22h 11min 59s, sys: 25min 49s, total: 1d 22h 37min 48s
Wall time: 2h 2min 14s


In [43]:
df_pred_all.head()

38       110       113       114       134       171  \
session_id                                                               
663721      0.111619  0.014616  0.015819  0.393292  0.083407  0.087782   
663761      0.265000  0.084178  0.447807  0.507279  0.341237  0.210086   
663763      0.180723  0.047931  0.074203  0.835555  0.479714  0.006166   
663775      0.058852  0.015049  0.014091  0.160487  0.310559  0.174879   
663778      0.134828  0.010294  0.093365  0.160464  0.136407  0.054237   

                 172       173       376       435       467       537  \
session_id                                                               
663721      0.001176  0.113097  0.105247  0.024853  0.023249  0.002273   
663761      0.453894  0.038657  0.278671  0.007666  0.004545  0.012603   
663763      0.006637  0.003068  0.213154  0.062969  0.042714  0.001345   
663775      0.018015  0.587651  0.041113  0.242624  0.034825  0.005434   
663778      0.009909  0.106057  0.060482  0.050241  0.008761  0.028959   

                 539       629       768  
session_id                                
663721      0.040888  0.055225  0.024540  
663761      0.267137  0.000339  0.327151  
663763      0.016071  0.009707  0.074509  
663775      0.047904  0.007011  0.203245  
663778      0.036203  0.001514  0.083410

In [44]:
df_score_all

38       110       113       114       134       171       172  \
0   0.826247  0.794000  0.802179  0.758762  0.745375  0.868761  0.870396   
3   0.823540  0.783918  0.796281  0.754738  0.739378  0.862179  0.862910   
5   0.822598  0.774979  0.789516  0.749585  0.737184  0.855232  0.855578   
10  0.820696  0.767330  0.786640  0.756155  0.744682  0.842316  0.833253   

         173       376       435       467       537       539       629  \
0   0.864301  0.754959  0.868450  0.869429  0.903415  0.858014  0.914694   
3   0.858000  0.751301  0.854208  0.860606  0.899014  0.849882  0.909442   
5   0.852917  0.748855  0.848399  0.855979  0.891379  0.841696  0.894157   
10  0.847202  0.758960  0.837073  0.836543  0.877262  0.830870  0.871031   

         768  
0   0.747212  
3   0.735356  
5   0.731096  
10  0.730911

In [45]:
cv = df_score_all.mean(axis=1)
print(cv)
print('- cv =', cv.mean())

0     0.829746
3     0.822717
5     0.816610
10    0.809395
dtype: float64
- cv = 0.8196170280400412


In [46]:
assert len(df_pred_all) == len(test)

In [47]:
submission = pd.merge(test[['session_id']], df_pred_all.reset_index(), on='session_id', how='inner')
assert len(submission) == len(test)

In [48]:
submission.drop(columns='session_id').to_csv('../outputs/submission.csv', index=False)

#### baseline_9-1: 'feature_fraction': 0.6,'bagging_fraction': 0.6,'bagging_freq': 2,
- feat = 333
- Wall time: 2h 2min 14s
- cv = 0.81961 (0.829/0.822/0.816/0.809)
- LB = 0.7658

#### baseline_9: 来店間隔、回数、曜日x時間の来店者数
- feat = 333
- Wall time: 2h 9min 51s
- cv = 0.82118 (0.831/0.824/0.818/0.810)
- LB = 0.7665

#### baseline_8: クーポン発行とターゲットの購買実績
- feat = 344
- Wall time: 2h 5min 13s
- cv = 0.820398 (0.832/0.822/0.817/0.809)
- LB = 0.7484

#### baseline_7: userごとのカテゴリの過去の購買実績
- feat = 329
- Wall time: 1h 59min 16s
- cv = 0.82148 (0.832/0.824/0.818/0.810)
- LB = 0.7671

#### baseline_6: sesssion内のターゲットの購買
- feat = 89
- Wall time: 37min 47s
- cv = 0.66015
- LB = 0.6247

#### baseline_5
- feat = 74
- Wall time: 39min 18s
- cv = 0.66859
- LB = 0.6230